spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -130,11 +130,18 @@ public Blob getGoogleStorageObject() throws IOException {
 	 */
 	public URL createSignedUrl(TimeUnit timeUnit, long timePeriods) throws IOException {
 		Blob blob = this.getGoogleStorageObject();
+
 		if (blob == null) {
-			return null;
+			if (!this.createBlobIfNotExists) {
+				return null;
+			}
+
+			createBlob();
```

Comment:
createBlob() returns the Blob, so I think you could just reuse  the blob reference from above.

Reply:
I'd have to call `blob = this.getGoogleStorageObject()` so that the object metadata is updated.

I thought that was a little unsightly and went for this one instead, but let me know if you have other better alternatives.